### PR TITLE
Update API -> complete becomes fit_transform

### PR DESCRIPTION
**Pull request by sergeyf**
Not too much going on here, but I did add some commentary to `README.md`.

I also deleted test_knn_timings.py. It was timing things that don't exist anymore.

@iskandr Do you have time to make `KNN` inductive? Should be pretty easy?

### CODE REVIEW

**Comment by sergeyf** [comment]
@iskandr I don't really feel like I need much of a review here. I'll wait a week and then merge.

File: fancyimpute/solver.py [hunk 1]
```diff
@@ -188,3 +196,29 @@ def complete(self, X):
...
+    def fit(self, X, y=None):
+        """
+        Fit the imputer on input `X`.
+
+        Note: all imputations should have a `fit_transform` method,
+        but only some (like IterativeImputer) also support inductive mode
+        using `fit` or `fit_transform` on `X_train` and then `transform`
+        on new `X_test`.
+        """
+        raise ValueError("%s.fit not implemented! This imputation algorithm likely"
```
Comment (ibarshai) [inline fancyimpute/solver.py]: Perhaps add "Use %s.fit_transform instead." at the end of error message.

File: fancyimpute/solver.py [hunk 2]
```diff
@@ -188,3 +196,29 @@ def complete(self, X):
...
+    def transform(self, X, y=None):
+        """
+        Transform input `X`.
+
+        Note: all imputations should have a `fit_transform` method,
+        but only some (like IterativeImputer) also support inductive mode
+        using `fit` or `fit_transform` on `X_train` and then `transform`
+        on new `X_test`.
+        """
+        raise ValueError("%s.transform not implemented! This imputation algorithm likely"
```
Comment (ibarshai) [inline fancyimpute/solver.py]: Perhaps add "Use %s.fit_transform instead." at the end of error message.

File: fancyimpute/solver.py (hunk 1 above)
Comment (sergeyf) [reply fancyimpute/solver.py]: I think the problem is that you NEED `fit` and then `transform` if you have training and then test data. So `fit_transform` is not an actual substitute. Maybe I can see "Only %s.fit_transform is supported at this time."?

**Comment by sergeyf** [comment]
@iskandr I asked Ilya to join our party and review stuff, so gonna merge.

**Pull request merged**